app/services/pseudonym.py

```diff
@@ -33,36 +32,31 @@ def __init__(
         )
         self._provider_id = provider_id
 
-    def submit(self, data: PseudonymCreateDto) -> PseudonymModel:
-        if data.provider_id is None:
-            data.provider_id = self._provider_id
-
-        logger.info(f"Exchanging BSN for provider {data.provider_id}")
+    def submit(self, data: PseudonymRequest) -> OprfPseudonymJWE:
+        """
+        Request OPRF blinded JWE from the pseudonym service.
+        """
+        logger.info(f"Request OPRF JWE for organisation {data.recipient_organization} with scope {data.recipient_scope}")
 
         try:
             response = self.http_service.do_request(
                 method="POST",
-                sub_route="register",
-                data={
-                    "provider_id": str(data.provider_id),
-                    "bsn_hash": data.bsn.hash(),
-                },
+                sub_route="oprf/eval",
+                data=data.model_dump()
             )
         except Exception as e:
-            logger.error(f"Failed to exchange BSN for pseudonym: {e}")
-            raise PseudonymError(f"Failed to exchange BSN for pseudonym: {e}")
+            logger.error(f"Failed to request OPRF pseudonym: {e}")
+            raise PseudonymError("Failed to request OPRF pseudonym") from e
 
         if response.status_code not in [201, 200]:
             raise PseudonymError(f"Failed to exchange BSN for pseudonym: {response.status_code}")
 
-        response_data = response.json()
         try:
-            new_pseudonym = Pseudonym(response_data.get("pseudonym", ""))
+            response_data = response.json()
+            return OprfPseudonymJWE(jwe=response_data.get("jwe"))
         except ValueError:
             raise PseudonymError("Failed to exchange BSN for pseudonym: invalid pseudonym")
 
-        return PseudonymModel(pseudonym=str(new_pseudonym))
-
     def request_reversible(self, dto: ReversiblePseudonymRequest) -> PseudonymModel:
```

Comment:
Remove in different PR,  `request_reversible` as authorization check is not needed anymore.